distributed/tests/test_worker.py

```diff
@@ -720,17 +720,18 @@ async def test_clean_nbytes(c, s, a, b):
 
 
 @pytest.mark.parametrize("as_deps", [True, False])
-@gen_cluster(client=True, nthreads=[("", 1)] * 20)
+@gen_cluster(client=True, nthreads=[("", 1)] * 21)
 async def test_gather_many_small(c, s, a, *workers, as_deps):
-    """If the dependencies of a given task are very small, do not limit the
-    number of concurrent outgoing connections
+    """If the dependencies of a given task are very small, do not limit the number of
+    concurrent outgoing connections. If multiple small fetches from the same worker are
+    scheduled all at once, they will result in a single call to gather_dep.
     """
     a.total_out_connections = 2
     futures = await c.scatter(
         {f"x{i}": i for i in range(100)},
         workers=[w.address for w in workers],
     )
-    assert all(w.data for w in workers)
+    assert all(len(w.data) == 5 for w in workers)
```

Comment:
1. I feel this is too granular for this test. Does this test require the distribution to be homogeneous? I think we should not test a "decide_worker" logic in this test.
2. How is `5` even a correct number when we're scattering 100 tasks to 21 workers?

Reply:
> I feel this is too granular for this test.

Ok, relaxing it.

> How is 5 even a correct number when we're scattering 100 tasks to 21 workers?

No, we're scattering 100 tasks to 20 workers.
See function signature: `c, s, a, *workers`

Reply:
Renamed workers to snd_workers for clarification